packages/grid/src/vaadin-grid.js

```diff
@@ -776,9 +776,20 @@ class Grid extends ElementMixin(
     const cellContent = document.createElement('vaadin-grid-cell-content');
     cellContent.setAttribute('slot', slotName);
 
+    const isRowHeader = column && column.rowHeader;
+
+    if (isRowHeader) {
+      tagName = 'th';
```

Comment:
Yes, true, if you check my comment with screenshots yesterday with my test results, I mentioned the sideeffect bolding part:

> I would assume it is okay based on the accessibility developer guide com:
> - https://www.accessibility-developer-guide.com/examples/tables/multiple-headers/#marking-multiple-cells-as-headers
>
> there is a related codepan example there (dynamic so cannot post the link of it)
> As said the related codepen example has similar style**s (bold for the row-header cells):**

The accessibility developer guide page asks for `<th>`:
- https://www.accessibility-developer-guide.com/examples/tables/multiple-headers/#marking-multiple-cells-as-headers

Also, the customer asked for the <th> tag. Jaws is a screen reader program I assume?  
Not sure if the `role` only is good enough for the customer, need to ask.

Good point on the dynamically toggleable thing, good insight!

Is there any way to have with `<th>` if needed?

Reply:
Note that if we take the `<th>` approach, we also need to handle:
- inverting the visual appearance implication (I'd assume, since there was no graphical design involved)
- fix CSS and JS files that have made assumptions about the tag names (and write relevant tests)
- pray that we won't face flow-component timing issues where the column cells get created _before_ the rowHeader property actually gets applied

If the ["role=rowheader"](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/rowheader_role)-approach works with all relevant cases, there's no good justification not to use it.

Reply:
The role="rowheader" option was evaluated as part of evaluating this feature, and it turns out Safari does not support it, which is why the AC specifies using the `<th>` element specifically.

Reply:
So it seems like we need to go with the th route:

- Why do we need to invert the visual appearance? If someone uses this feature, they should be familiar with the visual implications, it is happening if you add to any table. So it can be the responsibility of the developer, or maintainer of this code to have any styling they want for these lines?

- What other changes do we need to close this PR and use this <th> approach? And move to <flow-components> part?  So not sure about this part: `fix CSS and JS files that have made assumptions `

Reflected on the 1st, and praying for the 3rd of your point Tomi :).

Reply:
> turns out Safari does not support it

I'm not getting the row header announced on Safari with the `<th>` approach either.

<img width="937" alt="Screenshot 2023-06-21 at 11 02 45" src="https://github.com/vaadin/web-components/assets/1222264/9fb6dc3a-a133-44d5-8d35-8b99e2be8f94">

Reply:
Did you add the `scope` attrib? `<th scope="row">`

Reply:
> Why do we need to invert the visual appearance? If someone uses this feature, they should be familiar with the visual implications, it is happening if you add to any table.

Grid has Lumo-specific styles for everything and it intentionally overrides browser-specific default styling of the native `<table>` element. For example, the header row cells would also be bold by default, but their styling is overridden by Lumo Grid [styles](https://github.com/vaadin/web-components/blob/be619af7f7658fd5b1388daad2fb2b578a8c20ea/packages/grid/theme/lumo/vaadin-grid-styles.js#L202-L203).

Reply:
> * Why do we need to invert the visual appearance? If someone uses this feature, they should be familiar with the visual implications, it is happening if you add to any table. So it can be the responsibility of the developer, or maintainer of this code to have any styling they want for these lines?

Good question. I had not imagined we'd have bold font for the `<th>`s, as this is intended mainly for screen readers, and we _generally_ override most native styling for consistency, but in this case I don't really have a strong opinion.

Reply:
> Did you add the scope attrib? <th scope="row">

This PR does add the "scope" attribute also

Reply:
Also note that role="gridcell" needs to be omitted as it otherwise overrides the native <th> role.

Reply:
It's also covered [here](https://github.com/vaadin/web-components/pull/5843/files#diff-abb4f44539c9d46f2924e350662496e485a66fc0fc85dd22c577d9c30aebc1a2R788-R792)

Reply:
Thank you to both of you for insights, and comments.
How to progress now?:)
- Should we push this <th> route? (as Safari can be a blocker to all customers not just mine who sponsoring it).
- Plus what can be the reason that it is not working in Safari?

I think these are the open questions for now, plus how much work is needed to finish if we choose the <th> option.

@rolfsmeds I would think that is covered here:
```
  if (isRowHeader) {
      cell.setAttribute('scope', 'row');
    } else {
      cell.setAttribute('role', tagName === 'td' ? 'gridcell' : 'columnheader');
    }
 ```
 Did not tested, but remember it was not added, maybe we can explicitly remove as well if you guys think.

Reply:
My bad, now I see we did write the same with @tomivirkki about the `gridcell` part.

Reply:
The Safari problem is weird since I did get it to work with `<th scope="row">` back when this was evaluated, which I why I wrote those details into the AC.

My guess would be that there's some CSS that affects how it's interpreted.

I can try to find time to do a bit of quick re-testing tomorrow.

Reply:
Ok tested already now. Weirdly enough, now I can't get it to work at all – even with a plain HTML table like here: https://interactive-examples.mdn.mozilla.net/pages/tabbed/th.html

Reply:
Indeed, weird, maybe something changed within Safari between your previous test and, now it is broken? (I mean it was probably different Safari versions)

So it seems like Safari is fully broken for this feature? :S 
In that case, we can use the only `scope='row'` solution if the customer agrees? What do you think?

Reply:
Ah, actually, I was wrong(ish): it does work with that native HTML table, but the row-header cells are only announced when you switch rows (not when you traverse cells within the same row). It seems a bit hit-and-miss, but mostly does seem to work:
<img width="611" alt="image" src="https://github.com/vaadin/web-components/assets/4611559/b7645728-9304-4bc7-9f31-546fc23ab0e5">

So, we're back at square one.

Reply:
Tested this PR in Safari 16.5.1 (the latest one) and could not get VoiceOver working with `<th scope="row">`: for some reason, it isn't possible to reach `<th>` elements by arrow navigation.

Reply:
@web-padawan: thank you, there will be a new implementation for this feature soonish. We had a discussion with t with @rolfsmeds  and @tomivirkki, before they went on vacation (the related Acceptance Criteria were already updated). 

I wanted to get back to this week Tue-Wed, but I did not find the time to finish it yet... 
I am planning to wrap up, and change this PR ASAP, and let you know, then your review and approval would be more than welcome.

Reply:
Yes, @tomivirkki found that latest Safari+VO (on MacOS Ventura) seems to work just fine with `<td role="rowheader"...>` but not with `<th scope="row"...>` so we decided to change course to that approach instead. ([AC here](https://github.com/vaadin/platform/issues/3738) updated accordingly).